### PR TITLE
Add tokio bindings

### LINKED ISSUE #20: Possible tokio integration?
With the Futures and Async/Await features soon to be stabilized, is it possible to use this library asynchronously with Tokio? If so, could documentation and possibly a feature be added that allows for it so that actions to be taken after a sound is played can be done through future combinators? I haven't really dug into the library, but it didn't give us compilation issues, so we went with it for a hackathon. We'd like to integrate it into our production application, but we use a Tokio based framework and having this work with that natively would be really useful.

### PR DESCRIPTION
**Pull request by danieldg**
Fixes #20

### CODE REVIEW

**Comment by acheronfail** [comment]
While I love the idea of upstreaming this awesome tokio PulseAudio mainloop into `libpulse_binding`, would you be open to publishing a separate crate for others (like me!) to be able to depend on it and use it?

From my perspective - and correct me if I'm wrong @jnqnfe - this crate seems to be more of a 1:1 binding between PulseAudio's APIs rather than "a bunch of rust tools and integrations for PulseAudio". So to me, a separate crate makes sense for what is essentially "a tokio implementation of a PulseAudio mainloop".

TL;DR: I would love it if the `libpulse-tokio` crate added in this PR could be published to crates.io. Whether that's by upstreaming here, or by publishing it separately.

**Comment by danieldg** [comment]
@acheronfail - I just published the current version of this branch. While it still has a few limitations, those can be fixed in future versions. I'd be open to adding the maintainers here to the crate if it was merged, so it could be maintained/released in lock-step.

**Comment by MaxVerevkin** [comment]
> I just published the current version of this branch.

On crates.io? I couldn't find it.

**Comment by danieldg** [comment]
https://crates.io/crates/libpulse-tokio

**Comment by MaxVerevkin** [comment]
Thank you, I don't know why neither crates.io nor lib.rs could find it :(

**Comment by acheronfail** [comment]
@jnqnfe I was wondering if it'd be possible for this to be added to `libpulse-binding`?

I've been upgrading my dependencies, but upgrading from `2.28.2` to `2.30.1` actually means that @danieldg's `libpulse-tokio` breaks. :cry: 

Would be nice to have this all included in `libpulse-binding` itself. I can't remember if there was anything specific from blocking this?
